pkg/features/list.go

```diff
@@ -72,4 +72,7 @@ var (
 
 	// SensorDeploymentBuildOptimization enables a performance improvement by skipping deployments processing when no dependency or spec changed
 	SensorDeploymentBuildOptimization = registerFeature("Enables a performance improvement by skipping deployments processing when no dependency or spec changed", "ROX_DEPLOYMENT_BUILD_OPTIMIZATION", true)
+
+	// ScannerV4 enables various capabilities associated with ACS + Clair consolidated scanner.
+	ScannerV4 = registerFeature("Enable Scanner v4", "ROX_SCANNER_V4", false)
```

Comment:
Could someone please explain to me how the feature flag ROX_SCANNER_V4 is defined in the chart and how it is translated into each deployment? Is it an env var? 

Related to Ross comment, if env var, we don't need `ROX_SCANNER_V4_ENABLED` -- that was introduced to make Sensor aware that there is a Scanner V4 and perform local scans using it. If the Helm feature flags are exposed to all deployments as an env var, then we don't need a new one.

Reply:
What change for sensor are you referring to, @johannes94?

Reply:
I meant this specific comment from @jvdm:

> Related to Ross comment, if env var, we don't need ROX_SCANNER_V4_ENABLED -- that was introduced to make Sensor aware that there is a Scanner V4 and perform local scans using it. If the Helm feature flags are exposed to all deployments as an env var, then we don't need a new one.

Sounds to me like once we make the changes to secured cluster helm chart we need to make sure the variable is set for sensor deployment.

Reply:
That's correct. There is already an env var for that in Sensor Configuration: `ROX_SCANNER_V4_ENABLED`. Setting that var to `true` would make Sensor try to talk to Scanner V4 instead of V2. But the var name differs slightly from the name used in the Helm charts. It might be worthwhile changing the Sensor variable to use the same name. Central will also probably benefit from the the same variable.

Reply:
Thank you to @johannes94 for the explanation. Based on the information provided, it appears that the feature flag used during Helm chart rendering cannot be controlled by the customer in the same way as helm values (which we would specify with `--set`). This means that once we release our charts with the feature flag turned on, both Central and Secured Clusters will deploy Scanner V4 and Scanner V2.

However, we need to consider how to help customers disable Scanner V4. By "disable," we don't necessarily mean deleting the deployments, but rather configuring Central and Sensor to use or not use it. The simplest solution I can think of is to use environment variables in these services.

In any case, currently, there is already an environment variable for enabling Scanner V4 scans called `ROX_SCANNER_V4_ENABLED` in Sensor. Setting this variable to `true` makes Sensor attempt to communicate with Scanner V4 instead of V2. If `false` the current behavior is maintained, and Sensor will try to reach Scanner V2.

Reply:
Clarification: When we produce Helm charts with the feature flag `ROX_SCANNER_V4` being turned on, then this does not mean (as you write @jvdm):

> both Central and Secured Clusters will deploy Scanner V4 and Scanner V2.

Instead, it means:

> both Central and Secured Clusters (charts) will **be able to** deploy Scanner V4 and Scanner V2.

Whether Scanner V4 will actually be deployed depends on the setting of the Helm value `scannerV4.disable`. By default we have `disable=true`, i.e. users have to opt-in into Scanner V4.

Does this make sense?

Reply:
> Does this make sense?

It does! Thank you for clarifying that. In that case, setting an environment variable in Sensor to indicate that Scanner V4 is enabled is needed. Potential names are:

1.  `SCANNER_V4_ENABLED=true`: Current existing variable in Sensor.
2. `SCANNER_V4=true`: The feature flag in Helm.
3.  `SCANNER_V4_DISABLED=true`: to match the Helm value.

Setting the variable will instruct Sensor to reach to Scanner V4 instead of V2, and use Scanner V4 protos to send the Image Scans to Central -- so Central needs to have Scanner V4 for the image scan to work.

CC @dcaravel @RTann

Reply:
let's just use what we already have

Reply:
correct

Reply:
We may want to give this a bit more thought:

My current proposal for integration is to show/hide certain UI elements and perform some backend actions only if `scanner-v4` components are installed.  Because the UI can query feature flags from the Central API, my plan was to leverage a `changeable` feature flag for this - which would be updated via an env variable by the helm charts based on `scanner-v4` being enabled/disabled.

`ROX_SCANNER_V4` is currently a `changeable` feature, and would make sense to use for this purpose, however, because the value is changeable using it during meta-templating could yield some odd behaviors (such as: for meta-templating I'd want the feature `enabled` to give users the option to install `scanner-v4`, but central should only have the feature `enabled` if  the `scanner-v4` components were installed)

As a result, we may want to consider only using `unchangeable` features during `meta-templating` - and add a different feature flag for runtime decisions (such as hiding UI elements).

So perhaps we change `ROX_SCANNER_V4` to be an `UnchangeableFeature`, and convert `ROX_SCANNER_V4_ENABLED` to a `changeable` feature flag that will be used by both Central and Sensor, and set via an env var.

Reply:
Hope it's fine that I mark this as resolved.